lib/bamboo/mailer.ex

```diff
@@ -62,21 +62,22 @@ defmodule Bamboo.Mailer do
 
   defmacro __using__(opts) do
     quote bind_quoted: [opts: opts] do
-      config = Bamboo.Mailer.parse_opts(__MODULE__, opts)
-
-      @adapter config.adapter
-      @config config
-
       @spec deliver_now(Bamboo.Email.t) :: Bamboo.Email.t
       def deliver_now(email) do
-        Bamboo.Mailer.deliver_now(@adapter, email, @config)
+        config = get_config
+        Bamboo.Mailer.deliver_now(config.adapter, email, config)
       end
 
       @spec deliver_later(Bamboo.Email.t) :: Bamboo.Email.t
       def deliver_later(email) do
-        Bamboo.Mailer.deliver_later(@adapter, email, @config)
+        config = get_config
+        Bamboo.Mailer.deliver_later(config.adapter, email, config)
       end
 
+      otp_app = Keyword.fetch!(opts, :otp_app)
+
+      defp get_config, do: Bamboo.Mailer.parse_opts(__MODULE__, unquote(otp_app))
```

Comment:
I think I'd have to see it first, but it sounds pretty good. Definitely worth a try